Unit-01/04-flask-crud/app.py

```diff
@@ -1,3 +1,60 @@
-from flask import Flask
+from flask import Flask, render_template, request, redirect, url_for, abort
+from snack import Snack
+from functions import find_snack
+from flask_modus import Modus
 
-snack_list = []
\ No newline at end of file
+app = Flask(__name__)
+modus = Modus(app)
+
+snacks = [Snack('kit-kat', 'unhealthy')]
+
+@app.route("/")
+def root():
+	return redirect(url_for('index'))
+
+#To read snacks (GET /snacks)
+#To create snack (POST /snacks)
+@app.route("/snacks", methods=["GET", "POST"])
+def index():
+	if(request.method == "POST"):
+		snacks.append(Snack(request.form.get("name"), request.form.get("kind")))
+		return redirect(url_for('index'))
+	return render_template("index.html", snacks=snacks)
+
+#To read form to create snack(GET /snacks/new)
+@app.route("/snacks/new")
+def new():
+	return render_template("new.html")
+
+#To read snack (GET /snacks/<int:id>)
+#To edit snack (PUT /snacks/<int:id>)
+#To delete snack (DELETE /snacks/<int:id>)
+@app.route("/snacks/<int:id>", methods=["GET", "PATCH", "DELETE"])
+def show(id):
+	found = find_snack(id, snacks)
+	if found == None:
+		abort(404)
+	if(request.method == b"PATCH"):
+		found.name = request.form.get("name")
+		found.kind = request.form.get("kind")
+		return redirect(url_for('show', id=found.id))
+	if(request.method == b"DELETE"):
+		snacks.remove(found)
+		return redirect(url_for('index'))
+	return render_template("show.html", snack = found)
```

Comment:
nitpicking here, but it's usually considered more "Pythonic" to not have a space between kwarg operator, e.g. `(snack=found)`